### DISCUSSION TITLE
Can Leaf query the source codes in an OHDSI CDM dbms?

### DISCUSSION THREAD

**Discussion by artgoldberg** [uwrit/leaf#398] [Q&A] [answered]
Hello Nic & Team

As you know, in OHDSI CDMs each clinical dbms record can have a standard concept code and a source concept code. See meta-examples in this [list of domains and vocabularies](https://ohdsi.github.io/TheBookOfOhdsi/StandardizedVocabularies.html#tab:vocabList) for standard and source codes.

My question is: can Leaf query the source codes in an OHDSI CDM dbms?

Thanks
Arthur

**Comment by ndobb** [comment, marked as answer]
Hi @artgoldberg,

Thanks for your question! Yes, as Leaf concept `WHERE` clauses are flexible, Leaf can be configured to use either the OMOP standard concept code or source code (or even incorporate the `concept_ancestor` table to include descendant coded concepts).

For example, at UW we host one OMOP Leaf instance which happens to use source codes, as can be seen here:
![image](https://user-images.githubusercontent.com/10424962/115755704-7aee1080-a352-11eb-8ad7-cf2b98cf85be.png)

One could instead of course use the standard concept by changing the `WHERE` clause from:

`@.condition_source_value = 'E11.22'`

to 

`@.concept_id = 45595797`

which would accomplish the same thing (though you'd likely use different database indexing strategies and so on, depending on your implementation).

Also, in our [leaf-scripts repo](https://github.com/uwrit/leaf-scripts/tree/master/omop/5.3) you can find examples of how we build the concept tree dynamically and assign OMOP `WHERE` clauses by script as well. (example: https://github.com/uwrit/leaf-scripts/blob/master/omop/5.3/5_vitals.sql#L71).

Best,
-nic

**Reply by artgoldberg** [reply]
Thanks @ndobb , that's very helpful.
